[ci] dashboard required check structurally fails on all dependabot PRs (empty CLOUDFLARE_API_TOKEN)

## Summary

The required **`dashboard`** check (`.github/workflows/cloudflare.yml`) can **never go green on a dependabot PR**, so every grouped dependency bump (#1804, previously #1766 / #1754) is blocked from auto-merge.

## Root cause (verified)

The `dashboard` job guards against forks:

```yaml
# .github/workflows/cloudflare.yml (~L39-41)
if: >-
  github.event_name != 'pull_request' ||
  github.event.pull_request.head.repo.full_name == github.repository
```

This skips **forked** PRs — but **dependabot** PRs use an *in-repo* branch, so the guard passes and the job runs. GitHub runs `dependabot[bot]` events **without access to regular Actions secrets**, so the `Set secrets` step (~L124) sees an empty `CLOUDFLARE_API_TOKEN`:

```
✘ [ERROR] In a non-interactive environment, it's necessary to set a CLOUDFLARE_API_TOKEN environment variable for wrangler to work.
❌ Failed to set dashboard worker secrets
##[error]Process completed with exit code 1.
```

The job exits **before the build/type-check completes**, so the required check fails for reasons unrelated to the dependency change. (Confirmed on #1804 run `27879437813`.)

## Impact

- All dependabot dependency PRs are permanently red on the required check → cannot auto-merge → dependency debt accumulates and must be merged manually with admin override.

## Recommended fix (owner decision — changes required-check semantics)

**Option A (smaller):** Gate the secret/deploy steps so dependabot PRs run **build + type-check only**, skipping the deploy path:
```yaml
- name: Set secrets
  if: github.actor != 'dependabot[bot]'
- name: Deploy preview
  if: github.event_name == 'pull_request' && github.actor != 'dependabot[bot]'
```

**Option B (cleaner):** Split the job — the secret-free **build + type-check** becomes the *required* `dashboard` check; **preview-deploy** moves to a separate, non-required job that skips for forks and dependabot. This keeps real build verification on dependency PRs while removing the deploy dependency from the merge gate.

Either unblocks dependency PRs. Not done autonomously because it changes required-check semantics and deploy/secret behavior.

---
*Filed by the automated agent-loop (CI / DX finding). Verified against `cloudflare.yml` and run 27879437813.*

--- a/.github/workflows/cloudflare.yml
+++ b/.github/workflows/cloudflare.yml
@@ -122,6 +122,14 @@ jobs:
           fi
 
       - name: Set secrets
+        # Dependabot PRs run with GitHub's isolated dependabot secret store, so
+        # the deployment secrets resolve empty and `set-secrets.ts --strict`
+        # exits 1 — structurally failing the required `dashboard` check on every
+        # dependency PR (#1809). Skip the secret/deploy/verify steps for
+        # dependabot; build + type-check above still verify the dependency bump.
+        # Production deploys (push to main) are unaffected — actor is never
+        # dependabot there.
+        if: github.actor != 'dependabot[bot]'
         env:
           CLOUDFLARE_API_TOKEN: ${{ secrets.CLOUDFLARE_API_TOKEN }}
           CLOUDFLARE_ACCOUNT_ID: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
@@ -154,7 +162,7 @@ jobs:
           fi
 
       - name: Deploy preview
-        if: github.event_name == 'pull_request'
+        if: github.event_name == 'pull_request' && github.actor != 'dependabot[bot]'
         env:
           CLOUDFLARE_API_TOKEN: ${{ secrets.CLOUDFLARE_API_TOKEN }}
           CLOUDFLARE_ACCOUNT_ID: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
@@ -168,6 +176,8 @@ jobs:
         run: bunx wrangler deploy --minify
 
       - name: Verify deployment
+        # Skip for dependabot PRs — no preview was deployed (see Set secrets).
+        if: github.actor != 'dependabot[bot]'
         env:
           CHM_API_KEY_SECRET: ${{ secrets.CHM_API_KEY_SECRET }}
         run: |

